src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/KeystoneComponent.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    public class KeystoneInsight
+    {
+        public string Title;
+        public string Summary;
+        public Solution Solution;
+        public DateTime StartDate;
+        public DateTime ExpiryDate;
+
+        public KeystoneInsight(string title, string summary, Solution solution=null, DateTime? startDate= null, DateTime? expiryDate = null)
+        {
+            Title = title;
+            Summary = summary;
+            Solution = solution;
+            StartDate = startDate ?? DateTime.UtcNow;
+            ExpiryDate = expiryDate ?? DateTime.MaxValue;
+        }
+    }
+
+    public static class ResponseKeystoneComponentExtension
+    {
+        public static DiagnosticData AddKeystoneComponent(this Response response, KeystoneInsight keystoneInsight)
+        {
+            try
+            {
+                if (!(keystoneInsight.Title != null && keystoneInsight.Summary != null))
```

Comment:
perhaps check for IsNullOrWhitespace() ?

Reply:
Yup, also will add a check for if expiry date is before current date.